docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,562 @@
+# Diamond workflow support
+
+A diamond workflow allows for a pipeline to scale out the generation of a single bundle.
+
+A diamond is composed of multiple splits; each split tries to write a set of files.
+Ideally, 2 splits do not write the same file.
```

Comment:
So that this serves as documentation PEP-style, I would embed a definition here by saying

> If 2 splits write the same file, it's called a conflict, and ideally conflicts do not occur.

or something to that effect.  That way, when the commit logic uses the term "conflict" it's totally clear what's being described.

Reply:
what is PEP style?

Reply:
[Python Enhancement Proposals](https://www.python.org/dev/peps/) (it turns out, had to refer to python.org to unacronymize) are the process for deciding things about Python.  even after decision-making, PEPs are often referred to by number, so i've said, "refer to PEP8" more times than i can remember due to the absence of `pyfmt`

Reply:
[572](https://hub.packtpub.com/why-guido-van-rossum-quit/) is another PEP, probably to live on in infamy